### PR TITLE
Minor website overhaul

### PR DESCRIPTION
**Pull request by TheRealPear**
Since last week, I was hoping to improve this website. As a bonus, I learned Jekyll at the same time. I've also tested my changes across various screen sizes (iPhone 12, Samsung Galaxy A54, iPad 5, generic 1080p screen, etc..) Here are the changelog for this pull request:
* Added a plugin to automatically generate a `sitemap.xml` file. This will be useful for Google and other search engines so it can properly index the website and make sure its discovery are up to date.
    * This is for consistency as https://pgm.dev/ has this feature as well.
* Links to blog posts no longer end with `.html`, making it more consistent to links to the rest of the website. However, this will break older links. We could use redirect rules if this is important.
* Added proper OpenGraph and Twitter Card metadata in html head. Discord uses this to figure out how an embed should appear whenever the website is linked.
    * The website title is removed for all pages that is not the homepage to avoid repeating the website name over again.
    * Blog posts will now have a large banner picture (plus a fallback if a specific image isn't defined)
    * Here's an example of these changes from Discord: A sample Discord embed using Overcast Community website
* Fixed descriptions being the same across every webpages, since that is terrible for search engines.
* The website should fit iPhone X-esque screens with notches better now.
* Updated Bootstrap 4 and related libraries to the latest minor version. Browser compatibility should about the same.
    * Fixed a bug with table content not appearing in dark mode on the Mapmaker page.
* Changed disclaimer in the footer to have a slightly bigger font size and be normal-cased in html, but upper-cased in CSS. This allows for better customization if it's ever needed in the future and less chance of someone's screenreader shouting at them.
* Added GitHub links to staff where it makes sense (admins, developers, map developers, etc.) in the staff page
* Fixed oversized text on the home page when visiting from mobile or tablet, same thing for the footer as well.
* Fixed announcement sidebar not fitting in the screen when visiting the blog list page from mobile.
* Minor grammatical correction throughout the website.
* Forced certain image sizes so the website layout won't get disrupted when it's in the middle of loading. This should prevent pages from jumping randomly. I've also added alt texts/ARIA labels where necessary.
* Made most of the websites consistent whenever the title, GitHub link, etc is changed.
* Blog posts now support reader mode on some browsers.
* Fixed dead links in some blog posts.

### CODE REVIEW

**Comment by Brottweiler** [comment]
Thanks for this PR, awesome work! Would you mind sharing some screenshots of the most notable visual changes?

**Comment by TheRealPear** [comment]
> Would you mind sharing some screenshots of the most notable visual changes?

Sure, I had to create a separate branch with some differences (this will not affect this pull request) to actually deploy to GitHub Pages. I'd recommend checking out https://therealpear.github.io/overcastcommunity-site/ and comparing it against https://oc.tc/.

My intention was to not make any significant visual changes, however, the majority of the changes are intended for people with smaller screen resolutions. Here are some screenshots (please excuse broken image/links, this is a Jekyll quirk that shouldn't show up in the final version):

# Before
Before homepage on iPhone 12
Before blog list page on iPhone 12
Before footer on iPad mini resolution


# After
Resized jumbotron text:
After homepage on iPhone 12
Hide announcement sidebar for small screens:
After blog page list on iPhone 12
Fixed the spacing between the footer on tablet screens:
After footer on iPad mini resolution
Using Safari's reader mode on a blog post. This feature also works on Firefox and Microsoft Edge:
Safari reader mode on a blog post

**Comment by Brottweiler** [comment]
@applenick @Indicardo Would you mind taking a quick look at this PR? I am sure it is all in order, just want some second opinions if you notice anything.

File: assets/img/vote/site-2.png
Comment (applenick) [inline assets/img/vote/site-2.png:1]: Fantastic! Thanks for fixing this.

File: _data/staff.yml
Comment (applenick) [inline _data/staff.yml:19]: Interesting choice! I personally don't mind my GitHub being linked, but hopefully that's the same for everyone else.

File: _includes/head.html
Comment (applenick) [inline _includes/head.html:30]: Curious, were there any major differences between the bootstrap versions? (Smart move, just curious what changed)

File: _pages/mapmaker.html
Comment (applenick) [inline _pages/mapmaker.html:1]: Great changes! Often forget we have a mapmaker page. Are there any additional commands on mapmaker you believe would be worth including?

File: _posts/2022-07-04-Moderator-Applications.md
Comment (applenick) [inline _posts/2022-07-04-Moderator-Applications.md:5]:
```suggestion
excerpt: Finally, a way for you to apply!
```

Believe that'd be the proper grammar. Fix the issue in the post as well too, thanks!

**Review by applenick — APPROVED** [review]
Excellent work on everything @TheRealPear! I left a few small comments but in general everything is an improvement 👍

**Review by Indicardo — APPROVED** [review]
lgtm

File: _includes/head.html
Comment (TheRealPear) [reply _includes/head.html:30]: It's essentially just quality of life and bug fixes, some which might be necessary to contributors that edit the website or the users visiting https://oc.tc (think better support for accessibility settings, etc.) Without going into more details, you can read [Bootstrap's changelogs starting from 4.1.3](https://github.com/twbs/bootstrap/releases?page=4). I was originally going to upgrade to Bootstrap 5, however, it would've required rewriting a majority of the website and it would drop support for Internet Explorer and older version of browsers.

File: _data/staff.yml
Comment (TheRealPear) [reply _data/staff.yml:19]: I hope so as well. I was able to find people's GitHub either from looking at their Discord profile or looking at prior commits in other OCC or PGM repositories. I figured it would make sense that people that are actively involved with OCC development should have a link to direct visitors to their official GH profile.

File: _pages/mapmaker.html
Comment (TheRealPear) [reply _pages/mapmaker.html:1]: No additional commands that I know of

**Comment by TheRealPear** [comment]
I've made a last minute change regarding blog posts. I've reorganized then so that it will appear under the `/blog` folder to the end user. For example, the latest blog post will now show up as https://oc.tc/blog/announcement/2023/04/07/Cinco-For-SrMod/ rather than the old link, which was https://oc.tc/announcement/2023/04/07/Cinco-For-SrMod.html.

I would also like to clarify my changes on how hyperlinking works. Instead of hardcoding paths like `/assets/img/404.png`, this will cause Jekyll to not use the correct link in some situations (such as forking the website and deploying it to your own GitHub pages or ignoring any page-specific permalinks in place). Unfortunately, GitHub Pages themselves are still using [an older version of Jekyll](https://pages.github.com/versions/), so I also have to add `{{ site.baseurl }}` in front of the new link tag `{% link /assets/img/404.png %}`. The good news is that this should not be a problem once the pull request is merged, it is only intended as an easier way for people to test and view the website from their fork and Jekyll will ignore the tag if the `baseurl` value is missing from `_config.yml`.

**Comment by applenick** [comment]
@TheRealPear Again, nice job on everything!

Though regarding the change to the `/blog/`, will this change break existing links or will they remain available?

File: _includes/head.html
Comment (applenick) [reply _includes/head.html:30]: Thanks! Good to know.

**Comment by Indicardo** [comment]
> As I mentioned in the first post, we could setup a redirect rule if it is necessary

I'd say it is, yeah. I wouldn't want to update any of the existing Discord announcements to reflect these changes.

**Comment by TheRealPear** [comment]
Old links should be working now. This shouldn't be necessary for any new blog posts after this pull request.

**Comment by Brottweiler** [comment]
I think we should still merge this. Are all the comments resolved?

**Comment by TheRealPear** [comment]
This branch became more of a mess than I anticipated. I may need to squash some of the commits or redo it from scratch.

**Comment by Indicardo** [comment]
> This branch became more of a mess than I anticipated. I may need to squash some of the commits or redo it from scratch.

No need to, we will squash all commits anyway.

**Comment by TheRealPear** [comment]
Once again, thank you for rebasing for me @Indicardo. I've re-added changes that were wiped in the rebase (probably a Git thing, not sure how I can prevent that in the future). Blog links are more simplified now (category are no longer part of the url) and I'm satisfied with the changes so far.

**Comment by Brottweiler** [comment]
Thanks for that last commit! I'll merge this soon, and if you have more changes we can make more pull requests.

**Pull request merged**
